### DISCUSSION TITLE
Most efficient way of updating one dimension of NDArray

### DISCUSSION THREAD

**Discussion by tklinchik** [deepjavalibrary/djl#2487] [General]
What is the most efficient way to update one value or one dimension in let's say 2D NDArray?
It appears that setting a value at [row, col] is allocating DirectByteBuffer for that single value and I can't seem to find a proper way of updating a single row or a single column with either ByteArray or other data structure.

On the other hand, I tried taking advantage of slicing and updating that slice with values from ByteArray for a single row or column however it appears a slice will create a copy and won't let me mutate it by reference.
Here is what I tried:
```
array.get(NDIndex(":, 10")).set(byteArray)
```
but that will create a copy of a column and update that instead of updating the underlying.

In short, I'm looking for managing frequent updates to a reasonably large matrices and need an efficient way of doing so. If I take a standard `NDArray.set(index, double[])` approach its leaking way too much off heap allocation and causes OOMs and `NDArray.set(index, number)` would result in even more allocations just smaller.

Any guidance would be appreciated.

**Comment by TalGrbr** [comment]
Bumping, was this solved?

**Comment by zachgk** [comment]
Calling `get` creates a copy. You would mutate by `array.set(NDIndex(":, 10"), manager.create(byteArray, shape))`. However, making updates using the NDArrays is slow. You generally want to focus on doing operations with array programming. If you can't, it may be better to do the logic you can't transfer entirely using Java arrays and then copy back to NDArray. If you are running into OOM, look at our [memory management doc](http://docs.djl.ai/docs/development/memory_management.html)
